### PR TITLE
Fire auth modal

### PR DESCRIPTION
**Pull request by AD121390**
I know this took some time to finish but hopefully it's good. I spent a lot of time integrating everything together. Hopefully there aren't any bugs, but please take a look. All registered users are added to our firebase. Here is the link to look at that part of the dashboard. 

https://console.firebase.google.com/project/magic-kitchen-b5f37/authentication/users

You can also delete users from here. You also need to refresh the dashboard to see the new users.

### CODE REVIEW

File: src/components/SignupModal.jsx [hunk 1]
```diff
@@ -6,31 +6,246 @@ import {
...
+  const { isOpen: isLoginOpen, onOpen: onLoginOpen, onClose: onLoginClose } = useDisclosure()
+  const { isOpen: isSignUpOpen, onOpen: onSignUpOpen, onClose: onSignUpClose } = useDisclosure()
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [conf_password, setConfPassword] = useState('');
+  const [error, setError] = useState('');
+  const navigate = useNavigate();
+  const [user, setUser] = useState("");
+
+  useEffect( () => {
```
Comment (hetd54) [inline src/components/SignupModal.jsx:39]: are we sure that useeffect is needed here? what is the dependency that needs the re-render?

https://react.dev/reference/react/useEffect

File: src/components/SignupModal.jsx [hunk 2]
```diff
@@ -6,31 +6,246 @@ import {
...
+      window.location.reload();
+    }
+  })
+
+  const SignInGoogle = async () => {
+    try {
+      const user = await signInWithGoogle();
+      setUser(user.uid);
+      if (user) {
+        console.log("USER DETECTED");
```
Comment (hetd54) [inline src/components/SignupModal.jsx:52]: remove logs

File: src/components/SignupModal.jsx [hunk 3]
```diff
@@ -6,31 +6,246 @@ import {
...
+      setEmail("");
+      setPassword("");
+      navigate("/");
+    } catch (err) {
+      setError(`Authentication failed: ${err.message}`);
+    }
+  };
+
+
+  const validateRegisteration = () => {
```
Comment (hetd54) [inline src/components/SignupModal.jsx:97]: Registration misspelled here

File: src/components/SignupModal.jsx [hunk 4]
```diff
@@ -6,31 +6,246 @@ import {
...
+    let isValid = true;
+    if (password !== '' && conf_password !== ''){
+      if (password !== conf_password) {
+        isValid = false
+        setError('PASSWORDS MUST MATCH!')
+      } else if (!email.includes('@')) {
+          setError('Invalid Email')
+      }
+      else {
+        setError('Registeration Successful')
```
Comment (hetd54) [inline src/components/SignupModal.jsx:107]: Registration misspelled here

File: ui-debug.log
Comment (hetd54) [inline ui-debug.log:1]: let's remove this file

File: src/firebaseInit.js [hunk 5]
```diff
@@ -10,4 +16,21 @@ const firebaseConfig = {
 }
 
 // Initialize Firebase
-export const app = initializeApp(firebaseConfig)
+const app = initializeApp(firebaseConfig)
+export const auth = getAuth(app);
```
Comment (hetd54) [inline src/firebaseInit.js:20]: this should just be "getAuth()"

File: src/firebaseInit.js [hunk 6]
```diff
@@ -10,4 +16,21 @@ const firebaseConfig = {
 }
 
 // Initialize Firebase
-export const app = initializeApp(firebaseConfig)
+const app = initializeApp(firebaseConfig)
```
Comment (hetd54) [inline src/firebaseInit.js:19]: export this still, I think!

**Review by hetd54 — CHANGES_REQUESTED** [review]
Great first pass!  Be sure to also take a look at the behaviour of the navbar; the color changes after you sign in: 
Screenshot 2024-02-24 at 4 16 09 PM

**Comment by hetd54** [comment]
@AD121390 Could you also remove the "Submit Recipe" and "Account" links in the navbar while the "Signin/Signup" Modal is in the navbar?

File: src/components/SignupModal.jsx (hunk 1 above)
Comment (AD121390) [reply src/components/SignupModal.jsx:39]: That is how I check and see if there is a user that is logged in. Without this the Signup/login button does not update to the correct state.

**Comment by AD121390** [comment]
Yeah weird, the nav bar wasn't doing that before I committed v1. Must be from the main branch merge. I will look into hiding the account and submit recipe links

**Review by hetd54 — CHANGES_REQUESTED** [review]
Last thing needed that we discussed yesterday: the navbar links Account and Create Recipe should be hidden on the !logged state

**Review by T0ny-X — APPROVED** [review]
Online preview appears to be fine.

**Review by hetd54 — APPROVED** [review]
looks good now!

**Pull request merged**